src/rpc/ngHandlers/Subscribe.cpp

```diff
@@ -0,0 +1,194 @@
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <rpc/ngHandlers/Subscribe.h>
+#include <subscriptions/SubscriptionManager.h>
+
+namespace RPCng {
+
+boost::json::object
+SubscribeHandler::subscribeToStreams(
+    boost::asio::yield_context& yield,
+    std::vector<std::string> const& streams,
+    std::shared_ptr<WsBase> const& session,
+    std::shared_ptr<SubscriptionManager> const& manager) const
+{
+    boost::json::object response;
+    for (auto const& stream : streams)
+    {
+        if (stream == "ledger")
+            response = manager->subLedger(yield, session);
+        else if (stream == "transactions")
+            manager->subTransactions(session);
+        else if (stream == "transactions_proposed")
+            manager->subProposedTransactions(session);
+        else if (stream == "validations")
+            manager->subValidation(session);
+        else if (stream == "manifests")
+            manager->subManifest(session);
+        else if (stream == "book_changes")
+            manager->subBookChanges(session);
+    }
+    return response;
+}
+
+void
+SubscribeHandler::subscribeToAccounts(
+    std::vector<std::string> const& accounts,
+    std::shared_ptr<WsBase> const& session,
+    std::shared_ptr<SubscriptionManager> const& manager) const
+{
+    for (auto const& account : accounts)
+    {
+        auto const accountID = RPC::accountFromStringStrict(account);
+        manager->subAccount(*accountID, session);
+    }
+}
+
+void
+SubscribeHandler::subscribeToAccountsProposed(
+    std::vector<std::string> const& accounts,
+    std::shared_ptr<WsBase> const& session,
+    std::shared_ptr<SubscriptionManager> const& manager) const
+{
+    for (auto const& account : accounts)
+    {
+        auto const accountID = RPC::accountFromStringStrict(account);
+        manager->subProposedAccount(*accountID, session);
+    }
+}
+
+boost::json::array
+SubscribeHandler::subscribeToBooks(
+    std::vector<OrderBook> const& books,
+    std::shared_ptr<WsBase> const& session,
+    boost::asio::yield_context& yield,
+    std::shared_ptr<SubscriptionManager> const& manager) const
+{
+    boost::json::array snapshots;
+    std::optional<Backend::LedgerRange> rng;
+
+    for (auto const& internalBook : books)
+    {
+        if (internalBook.snapshot)
+        {
+            if (!rng)
+                rng = sharedPtrBackend_->fetchLedgerRange();
+            auto const getOrderBook = [&](auto const& book) {
+                auto const bookBase = getBookBase(book);
+                auto const [offers, _] = sharedPtrBackend_->fetchBookOffers(bookBase, rng->maxSequence, 200, yield);
```

Comment:
This `200` would be great to have as a meaningful constant somewhere in the class/function